### ISSUE TITLE
Failed to do failover in 3 node patroni postgres-11 cluster

### ISSUE THREAD

**Issue by sheeri185adidam** [zalando/patroni#1518]
I have a three node patroni postgres-11 cluster running.The patroni cluster settings are:

```
consul:
  url: <consul URL>

bootstrap:
  initdb:
    encoding: UTF8
  dcs:
    ttl: 30
    loop_wait: 10
    retry_timeout: 10
    maximum_lag_on_failover: 1048576
    postgresql:
      use_pg_rewind: true
      use_slots: true
      remove_data_directory_on_diverged_timelines: false
      remove_data_directory_on_rewind_failure: true
      parameters:
        archive_command: /bin/true
        archive_mode: 'on'
        archive_timeout: 1800s
        checkpoint_completion_target: 0.9
        default_statistics_target: 100
        effective_cache_size: 24GB
        effective_io_concurrency: 200
        hot_standby: 'on'
        log_directory: /var/log/postgresql
        log_filename: postgresql-%a.log
        logging_collector: 'on'
        maintenance_work_mem: 2GB
        max_connections: 2000
        max_parallel_maintenance_workers: 4
        max_parallel_workers: 8
        max_parallel_workers_per_gather: 4
        max_replication_slots: 10
        max_wal_senders: 10
        max_wal_size: 8GB
        max_worker_processes: 8
        min_wal_size: 2GB
        random_page_cost: 1.1
        shared_buffers: 8GB
        ssl: 'on'
        ssl_cert_file: /etc/ssl/certs/ssl-cert-snakeoil.pem
        ssl_key_file: /etc/ssl/private/ssl-cert-snakeoil.key
        track_commit_timestamp: 'on'
        wal_buffers: 16MB
        wal_keep_segments: 64
        wal_level: logical
        wal_log_hints: 'on'
        work_mem: 5242kB

postgresql:
  listen: 0.0.0.0:5432
  connect_address: <IP Address of the node>:5432
  data_dir: /var/lib/postgresql/11/<db name>
  pgpass: /tmp/pgpass0
  authentication:
    superuser:
      username: postgres
      password: ****
    replication:
      username: replicator
      password: ****
    rewind:
      username: rewind_user
      password: ****
  pg_hba:
    - local all all peer
    - host all all 127.0.0.1/32 md5
    - hostssl all postgres 0.0.0.0/0 md5
    - host all postgres 0.0.0.0/0 reject
    - host all all 0.0.0.0/0 md5
    - hostssl all postgres ::0/0 md5
    - host all postgres ::0/0 reject
    - host all all ::0/0 md5
    - local replication all peer
    - host replication all 127.0.0.1/32 md5
    - hostssl replication replicator 0.0.0.0/0 md5
    - hostssl replication replicator ::0/0 md5
    - host replication replicator 0.0.0.0/0 md5
    - host replication replicator 127.0.0.1/32 trust
    - local postgres   postgres   peer
    - host all all 0.0.0.0/0 md5
  create_replica_methods:
    - basebackup
  basebackup:
    max-rate: '250M'

tags:
  nofailover: false
  noloadbalance: false
  clonefrom: false
  nosync: false

restapi:
  connect_address: <IP Address of the node>:8008
  listen: 0.0.0.0:8008
```

When I shutdown the primary, I was hoping the synchronous standby would become primary. But instead on both the standby nodes I see the following error:

`INFO: following a different leader because i am not the healthiest node`

Not sure why synchronous standby did not become primary. I am using Patroni 1.6.1 on Ubuntu 18.04

**Comment by sheeri185adidam** [comment]
I tried another scenario where I shutdown one of the standby servers before shutting down primary so there was only one standby server running. Then I shutdown the primary and saw that the standby still wasn't promoted to primary. I looked at the postgresql logs in `/var/log/postgresql/postgresql-Thu.log`

```
2020-04-30 01:33:10.592 UTC [1304] LOG:  replication terminated by primary server
2020-04-30 01:33:10.592 UTC [1304] DETAIL:  End of WAL reached on timeline 14 at 0/A0000098.
2020-04-30 01:33:10.592 UTC [1304] FATAL:  could not send end-of-streaming message to primary: no COPY in progress
2020-04-30 01:33:10.592 UTC [1299] LOG:  invalid record length at 0/A0000098: wanted 24, got 0
2020-04-30 01:33:10.624 UTC [1480] FATAL:  could not connect to the primary server: server closed the connection unexpectedly
		This probably means the server terminated abnormally
		before or while processing the request.
2020-04-30 01:33:10.860 UTC [1296] LOG:  received fast shutdown request
2020-04-30 01:33:10.861 UTC [1296] LOG:  aborting any active transactions
2020-04-30 01:33:10.861 UTC [1308] FATAL:  terminating connection due to administrator command
2020-04-30 01:33:12.914 UTC [1485] FATAL:  the database system is shutting down
2020-04-30 01:33:12.915 UTC [1486] FATAL:  the database system is shutting down
2020-04-30 01:33:14.417 UTC [1488] FATAL:  the database system is shutting down
2020-04-30 01:33:14.418 UTC [1489] FATAL:  the database system is shutting down
2020-04-30 01:33:15.868 UTC [1301] LOG:  shutting down
2020-04-30 01:33:15.909 UTC [1296] LOG:  database system is shut down
2020-04-30 01:33:16.401 UTC [1498] LOG:  database system was shut down in recovery at 2020-04-30 01:33:15 UTC
2020-04-30 01:33:16.401 UTC [1498] WARNING:  recovery command file "recovery.conf" specified neither primary_conninfo nor restore_command
2020-04-30 01:33:16.401 UTC [1498] HINT:  The database server will regularly poll the pg_wal subdirectory to check for files placed there.
2020-04-30 01:33:16.401 UTC [1498] LOG:  entering standby mode
2020-04-30 01:33:16.404 UTC [1498] LOG:  consistent recovery state reached at 0/A0000098
2020-04-30 01:33:16.404 UTC [1498] LOG:  invalid record length at 0/A0000098: wanted 24, got 0
2020-04-30 01:33:16.404 UTC [1495] LOG:  database system is ready to accept read only connections
2020-04-30 01:33:16.406 UTC [1499] FATAL:  the database system is starting up
```

**Comment by sheeri185adidam** [comment]
I figured out the problem. For some reason the capital letters in the names of my nodes were converted to small letters so patroni failed to find the standby nodes. Like below:
`{"leader":"Pa-db1","sync_standby":"pa-db2"}`

The correct name of the node is `Pa-db2` and similarly the second standby node name is `Pa-db3`. I changed all the nodes names to small letters and failover worked!

**Comment by sheeri185adidam** [comment]
I figured out the problem. For some reason the capital letters in the names of my nodes were converted to small letters so patroni failed to find the standby nodes. Like below:
{"leader":"Pa-db1","sync_standby":"pa-db2"}

The correct name of the node is Pa-db2 and similarly the second standby node name is Pa-db3. I changed all the nodes names to small letters i.e. pa-db1, pa-db2, pa-db3 and failover worked!
